### PR TITLE
fix: use native `exists()` method in `GSClient`

### PR DESCRIPTION
**Pull request by bachya**
SSIA

Closes https://github.com/drivendataorg/cloudpathlib/issues/356

----------------

Contributor checklist:

 - [x] I have read and understood `CONTRIBUTING.md`
 - [x] Confirmed an issue exists for the PR, and the text `Closes #issue` appears in the PR summary (e.g., `Closes #123`).
 - [x] Confirmed PR is rebased onto the latest base
 - [x] Confirmed failure before change and success after change
 - [ ] Any generic new functionality is replicated across cloud providers if necessary
 - [x] Tested manually against live server backend for at least one provider
 - [ ] Added tests for any new functionality
 - [x] Linting passes locally
 - [x] Tests pass locally
 - [x] Updated `HISTORY.md` with the issue that is addressed and the PR you are submitting. If the top section is not `## UNRELEASED``, then you need to add a new section to the top of the document for your change.

### CODE REVIEW

File: cloudpathlib/gs/gsclient.py
Comment (pjbull) [inline cloudpathlib/gs/gsclient.py:172]: The `_exists()` is for all potential paths, not just buckets so I think the update is just for the short-circuit path of checking if the bucket exists.

Should be something like:

```python
if not cloud_path.blob:
    return self.client.bucket(cloud_path.bucket).exists()
    
return self._is_file_or_dir(cloud_path) in ["file", "dir"]
```

**Review by pjbull — CHANGES_REQUESTED** [review]
Thanks @bachya!

One change here. Can you also confirm that you tested the case in #356 with a totally empty bucket and this method works?

**Comment by bachya** [comment]
@pjbull Confirmed, I have this working on an empty GCP bucket.

**Comment by pjbull** [comment]
Thanks @bachya, happy to merge this once the code quality and mocked tests are passing.

**Comment by pjbull** [comment]
Tests need an update in the mocked version (which mock the cloud SDKs):
 - Update [`MockBucket` to have exists method.](https://github.com/drivendataorg/cloudpathlib/blob/c9da3f3ed7484f621f3ca33eef94ffcf0f0ca26c/tests/mock_clients/mock_gs.py#L120-L160)
 - Should be [implemented like the Azure backend to just be true for the test](https://github.com/drivendataorg/cloudpathlib/blob/c9da3f3ed7484f621f3ca33eef94ffcf0f0ca26c/tests/mock_clients/mock_azureblob.py#L142-L146).

More info on running the test suite here: https://cloudpathlib.drivendata.org/stable/contributing/#tests

**Comment by bachya** [comment]
Thanks, @pjbull. I couldn't see the results until you approved the CI run; got them now, and I'll update the tests.

**Comment by pjbull** [comment]
@bachya just fyi, you can run `make lint` and `make test` locally so you don't have to wait for the CI runs.

**Review by pjbull — APPROVED** [review]

**Comment by pjbull** [comment]
Looks good! I'm going to merge into a repo-local branch so we can run against the live server backends, which is not possible on PRs from forks.

**Pull request merged**
